crates/bevy_animation/src/lib.rs

```diff
@@ -482,26 +519,53 @@ impl AnimationClip {
         &mut self,
         target_id: AnimationTargetId,
         time: f32,
-        event: impl AnimationEvent,
+        event: impl Event + Clone,
     ) {
-        self.add_event_to_target_inner(AnimationEventTarget::Node(target_id), time, event);
+        self.add_event_fn_to_target(
+            target_id,
+            time,
+            move |commands: &mut Commands, entity: Entity, _time: f32, _weight: f32| {
+                commands.entity(entity).trigger(event.clone());
+            },
+        );
     }
 
     /// Add a untargeted [`AnimationEvent`] to this [`AnimationClip`].
     ///
-    /// The `event` will trigger on the [`AnimationPlayer`] entity once the `time` (in seconds)
+    /// The `func` will trigger on the [`AnimationPlayer`] entity once the `time` (in seconds)
     /// is reached in the animation.
     ///
+    /// For a simpler [`Event`]-based alternative, see [`AnimationClip::add_event`].
     /// See also [`add_event_to_target`](Self::add_event_to_target).
-    pub fn add_event(&mut self, time: f32, event: impl AnimationEvent) {
-        self.add_event_to_target_inner(AnimationEventTarget::Root, time, event);
+    pub fn add_event_fn(
+        &mut self,
+        time: f32,
+        func: impl Fn(&mut Commands, Entity, f32, f32) + Send + Sync + 'static,
```

Comment:
This signature is nasty for Rust beginners to read. A doc test would go a long way here.